src/connection.rs

```diff
@@ -948,18 +945,19 @@ async fn run_connection_supervisor(mut state: NSQDConnectionState) {
             warn!("drained {} messages", drained);
         }
 
-        if now.elapsed() >= state.config.shared.backoff_healthy_after {
-            info!("run_connection_supervisor resetting backoff");
+        if let Some(sleep_for) = backoff.next_backoff() {
+            info!(
+                "run_connection_supervisor sleeping for: {}",
+                sleep_for.as_secs()
+            );
+            tokio::time::sleep(sleep_for).await;
 
+        } else {
+            info!("run_connection_supervisor resetting backoff");
             backoff.reset();
         }
 
-        let sleep_for = backoff.next_backoff().unwrap();
-        info!(
-            "run_connection_supervisor sleeping for: {}",
-            sleep_for.as_secs()
-        );
-        tokio::time::sleep(sleep_for).await;
+        // under some circumstances it panics
```

Comment:
??

Reply:
Well, it's hard to reproduce consistently, but sometimes it seems that the check 
```rust
if now.elapsed() >= state.config.shared.backoff_healthy_after 
```
might return false if for example they differ by a few nanoseconds, and then the `backoff.next_backoff()` returns None, which gets unwrapped, I encountered it a few times when using `NSQConsumer`, but then again the occurrence is quite unpredictable.

Reply:
I'll remove comment